.github/workflows/spack_build.yml

```diff
@@ -0,0 +1,185 @@
+# https://spack.readthedocs.io/en/latest/binary_caches.html#spack-build-cache-for-github-actions
+name: Spack Builds (Ubuntu x86_64 Buildcache)
+
+on: [pull_request]
+
+env:
+  SPACK_COLOR: always
+  REGISTRY: ghcr.io/llnl
+  SPACK_CACHE: /opt/spack-cache
+  tempdir: /opt/spack-cache
+  TMP: /opt/spack-cache
+  TMPDIR: /opt/spack-cache
+  # Our repo name contains upper case characters, so we can't use ${{ github.repository }}
+  IMAGE_NAME: hiop
+  USERNAME: hiop-bot
+  BASE_VERSION: ubuntu-24.04-fortran
+
+jobs:
+  base_image_build:
+    runs-on: ubuntu-24.04
+    permissions:
```

Comment:
@nychiang tokens are automatically created by pipelines with permissions based on this YAML config.